### ISSUE TITLE
`Vi{` incorrectly selects the line with the closing bracket

### ISSUE THREAD

**Issue by KnorpelSenf** [VSCodeVim/Vim#9337]
## Description

In Vim, you can do `Vi{` (visual line inside block) to select all lines inside the block but not the lines that contain `{` and `}`.

This plugin incorrectly selects the line with the closing bracket.

## Reproduction

Write the following text.

```txt
Begin
{
  Block A
  Block B

  Block C
}
End
```

Move the cursor to any line inside the block, such as the empty line, and press `Vi{`.

### Expected

Only
```txt
  Block A
  Block B

  Block C
```
are selected.

### Actual

The closing bracket is highlighted, too, so the following text ends up in the selection.
```txt
  Block A
  Block B

  Block C
}
```

![grafik](https://github.com/user-attachments/assets/6c9945af-b109-4be9-bab2-d58d00e7c7e1)

**Environment (please complete the following information):**

- Extension (VsCodeVim) version: 1.28.1
- VSCode version: 1.95.0
- OS: Debian 12 (bookworm)

**Comment by karshPrime** [comment]
Yup error's persistent on my system as well

**Environment**
* Extension (VsCodeVim) version: 1.28.1
* VSCode version: 1.95.2
* OS: MacOS 15.1

**Comment by karshPrime** [comment]
I don't think this is an error. `V` is to select the whole line. If you want to select code inside the braces `{}`, then use `vi{` - that'd give you the functionality you're after.

**Comment by KnorpelSenf** [comment]
I'm not after selecting code inside the braces `{}`.

**Comment by KnorpelSenf** [comment]
> `vi{` and `Vi{` are two different things.

I know. The former works correctly, the latter doesn't.

`vi{` will take your cursor just before the final closing brace, giving you the behavior you are expecting. `Vi{` takes you cursor to the same position, but since `V` means it will select the entire line the cursor is on, it includes the final closing brace as well.

Can you tell me what happens when you try it in vim? Because on my machine, that is not what is happening.

I expect this extension to work like vim. Right now, it doesn't.

**Comment by Whiskas101** [comment]
You are correct, the selection terminates just before the newline character in vim or neovim, but it doesn't seem to do so here in vscodevim.

I was mistaken. I have only ever used `vi{` so I assumed the difference was intended behaviour.

**Comment by Whiskas101** [comment]
Do you have neovim path set up in the json config?

**Comment by KnorpelSenf** [comment]
I did not have it set up at the time I opened this issue. I didn't have neovim installed, either.

Now I do have neovim and I have stopped using this extension. The neovim integration is still labelled experimental and after all, I was looking for better stability. That's why I felt like https://github.com/vscode-neovim/vscode-neovim was a better fit. Can't say it's bug-free, either, but at least all the motions work correctly.

**Comment by KnorpelSenf** [comment]
@Whiskas101 the bugs in that other extension are getting worse … coming back to this one, I now have neovim and enabled it in the config. It is present in my PATH, and I also tried specifying it explicitly in the extension config file, but none of this affects the behaviour of `Vi{`. Note that if I use neovim directly, `Vi{` behaves correctly.

Are there any logs or other information that would help?

**Comment by Whiskas101** [comment]
That is expected, neovim executable is only used for certain commands like Ex

**Comment by KnorpelSenf** [comment]
Fascinating, I see. I have adjusted my muscle memory to switch between `Vi{` and `Vi{k` depending on my editor … it's not ideal but hey, I can live with it.

It presumably isn't terribly hard to fix this in the extension, but I sort of have too many things to do already so I don't think I'll contribute a fix, sorry.

**Comment by Whiskas101** [comment]
I'll attempt at finding a fix, I have some time on my hands these days.

**Comment by KnorpelSenf** [comment]
Epic, thank you so much!

**Comment by Whiskas101** [comment]
@KnorpelSenf I have opened a PR, you could test it out in the meantime

**Comment by KnorpelSenf** [comment]
Sorry for the delay, I have time to test the changes now. I see that they have already been merged but not released. How should I try this out? Should I still build it locally, or just wait for a release and report back?
